File: Data Structures & Algorithms/longest-common-prefix/submission-0.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        res = ""
        if not strs[0]:
            return res
        
        for i in range(len(strs[0])):
            print(strs[0][i])
            for s in strs:
                if i >= len(s):
                    return res
                
                if s[i] != strs[0][i]:
                    return res
            res += strs[0][i]

        return res
                

File: Data Structures & Algorithms/anagram-groups/submission-0.cpp
#include <vector>
#include <string>
#include <unordered_map>
#include <algorithm>

using namespace std;

class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {

        unordered_map<string, vector<string>> m;

        // Sort & insert into the map
        for(int i = 0; i < strs.size(); i++){
            string sortedS = strs[i];
            sort(sortedS.begin(), sortedS.end());
            m[sortedS].push_back(strs[i]);
        }

        vector<vector<string>> res;

        for(auto& v: m){
            res.push_back(v.second);
        }

        return res;
    }
};